drivers/hmis/app/graphql/types/forms/form_definition.rb

```diff
@@ -1,20 +1,103 @@
+# frozen_string_literal: true
+
 ###
 # Copyright 2016 - 2023 Green River Data Analysis, LLC
 #
 # License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
 ###
 
-# frozen_string_literal: true
-
 module Types
   class Forms::FormDefinition < Types::BaseObject
     description 'FormDefinition'
     field :id, ID, null: false
     field :role, Types::Forms::Enums::FormRole, null: false
     field :definition, Forms::FormDefinitionJson, null: false
 
+    # Filtering is implemented within this resolver rather than a separate concern. This
+    # gives us convenient to access the lazy batch loader for records (funder, orgs) that
+    # we might need to apply filter. Probably the filtering should get moved to it's own
+    # class down the road
     def definition
-      JSON.parse(object.definition)
+      # the col is jsonb... somehow we get a string here?
+      eval_items([object.definition])[0]
+    end
+
+    protected
+
+    # @param items [Array<Hash>]
+    # @return items [Array<Hash>]
+    def eval_items(items)
+      items.filter do |item|
+        if eval_rule(item['rule'])
+          if item['item']
+            # filter children
+            item['item'] = eval_items(item['item'])
+            # if all children are filtered out, exclude the parent
+            item['item'].any?
+          else
+            true
+          end
+        else
+          false
+        end
+      end
+    end
+
+    # @param rule [nil, Array, Hash]
+    # @return [Boolean]
+    def eval_rule(rule)
+      # if there's no rule, default to true
+      return true if rule.nil?
+
+      operator = rule.fetch('operator')
+      case operator
+      when 'EQUAL'
+        # { variable: 'projectType', operator: 'EQUAL', value: '1' }
+        eval_var(rule.fetch('variable')) == rule.fetch('value')
+      when 'INCLUDE'
+        # { variable: 'projectFunderIDs', operator: 'INCLUDE', value: '1' }
+        eval_var_multi(rule.fetch('variable')).include?(rule.fetch('value'))
+      when 'ANY'
+        # { operator: 'ANY', parts: [ ... ] },
+        rule.fetch('parts').any? { |r| eval_rule(r) }
+      when 'ALL'
+        # { operator: 'ALL', parts: [ ... ] },
+        rule.fetch('parts').all? { |r| eval_rule(r) }
+      else
+        raise "operator not supported: #{operator}"
+      end
+    end
+
+    # @param key [String]
+    # @return [String, Integer, nil]
+    def eval_var(key)
+      case key
+      when 'projectType'
+        project&.project_type
+      else
+        raise "unknown variable #{key}"
+      end
+    end
+
+    # @param key [String]
+    # @return [Array<String, Integer>]
+    def eval_var_multi(key)
+      case key
+      when 'projectFunderIDs'
+        project_funders.map(&:funder_id)
```

Comment:
```suggestion
        project_funders.map(&:funder)
```
Whoops, sorry this wasn't clear. It's the `Funder` that we're applying logic on, not the `FunderID`. The Funder ID is the "HUD ID" for the record, whereas the `Funder` is an integer from the 2.06.1 list.

<img width="662" alt="Screen Shot 2023-07-31 at 5 20 19 PM" src="https://github.com/greenriver/hmis-warehouse/assets/5333982/32398b17-f225-4e90-9cda-7eef6475c6c0">

https://files.hudexchange.info/resources/documents/HMIS-CSV-Format-Specifications-2024.pdf

Reply:
The way I'm testing in the application on my branch is: go to project => funder => add funder => select one from the list that's relevant (in my case YHDP which is 46) => enroll client or go to an existing enrollment and open the Intake assessment. Changing the funder, or the project type on the project, should update the assessment.


Feel free to pull in https://github.com/greenriver/hmis-warehouse/pull/3333/commits/26fde6ea6aebc5edd48bb9d380278ddb96e19711 here. This was working for me, and the C3 logic (and patch) I laid out here is correct to be merged. (Unlike income sources which was a fake example to get started).

Reply:
Sorry I misunderstood, thanks for clarifying! I renamed that variable from projectFunderIDs to projectFunders and changed it to expect an integer rather than a string (hud says it's an int but we seem to store it as a string).

I also merged in `gig/default-assmt-patches` and updated the variable name in the new rules